match2/commons/match_group_display_bracket.lua

```diff
@@ -1,738 +1,745 @@
-                    )
-                ):cssText('position: absolute;') or
-                ''
-        ):cssText('height:' .. maxHeight .. 'px;') or
-        ''
-
-    local out =
-        html.create('div'):addClass('brkts-match-wrapper'):css('display', 'block'):node(isend and '' or skiproundConnector):node(
-        isend and '' or
-            html.create('div'):addClass('brkts-line-container'):node(
-                html.create('div'):addClass('brkts-line-horizontal')
-            ):node(
-                html.create('div'):addClass('brkts-line-spacer brkts-line-spacer-upper'):cssText(
-                    'height:' .. self.padding .. 'px'
-                )
-            ):node(
-                html.create('div'):addClass('brkts-line-spacer brkts-line-spacer-lower'):cssText(
-                    'height:' .. self.padding .. 'px'
-                )
-            ):node(html.create('div'):addClass('brkts-line-horizontal'))
-    ):node(match):node(
-        (ishead and not self.qualwin) and '' or
-            html.create('div'):addClass('brkts-line-container'):node(
-                html.create('div'):addClass(
-                    (wassingle or self.qualwin) and 'brkts-line-horizontal-single' or 'brkts-line-horizontal'
-                )
-            )
-    ):cssText('margin-top:' .. _pxFix(marginTop) .. 'px')
-
-    out =
-        html.create('div'):addClass('brkts-header-wrapper' .. (bracketreset ~= nil and ' brkts-br-wrapper' or '')):css('display', 'block'):node(
-        header
-    ):node(out):node(thirdPlaceHeaderNode):node(thirdplacematch):cssText('height:' .. maxHeight .. 'px;')
-
-    out =
-        html.create('div'):addClass('brkts-round-wrapper' .. (wassingle and ' brkts-round-wrapper-single' or '')):css('align-items', 'flex-start'):node(
-        html.create('div'):addClass('brkts-round'):node(s1):node(s2)
-    ):node(
-        not isfull and '' or
-            html.create('div'):addClass('brkts-line-container2'):node(
-                html.create('div'):addClass('brkts-line'):cssText(
-                    'height:' .. lineheight .. 'px;margin-top:' .. upperinset .. 'px;'
-                )
-            ):node(
-                html.create('div'):addClass('brkts-line'):cssText(
-                    'height:' .. lineheight .. 'px;margin-top:' .. (2 * self.padding) .. 'px;'
-                )
+    if props.matchHasDetails(props.match) then
+        local matchSummaryNode = DisplayUtil.TryPureComponent(props.MatchSummaryContainer, {
+            bracketId = props.match.matchId:match('^(.*)_'), -- everything up to the final '_'
+            matchId = props.match.matchId,
+        })
+
+        local matchSummaryPopupNode = html.create('div'):addClass('brkts-match-info-popup')
+            :node(matchSummaryNode)
+
+        matchNode
+            :node(
+                html.create('div'):addClass('brkts-match-info-icon')
+                    :css('top', #props.match.opponents * props.opponentHeight / 2 - 6 - 1 .. 'px')
```

Comment:
That's a complicated sum. What's going on there?

Reply:
It aligns the (vertical) middle of the match with the middle of the 12px icon. The `-1` is for the top border, because the height includes top and bottom borders but the position is relative to the content box of the match.

I added a comment explaining this.